Detections/MultipleDataSources/SucessfullSiginFromPhingLink.yaml

```diff
@@ -0,0 +1,143 @@
+severity:	Medium
+requiredDataConnectors:
+  - connectorId:  OfficeATP
+    dataTypes:
+    - SecurityAlert
+  - connectorId:  PaloAltoNetworks
+    dataTypes:
+    - CommonSecurityLog (PaloAlto)
+  - connectorId:  Fortinet
+    dataTypes:
+    - CommonSecurityLog (Fortinet)
+  - connectorId:  CheckPoint
+    dataTypes:
+    - CommonSecurityLog (CheckPoint)
+  - connectorId:  Zscaler
+    dataTypes:
+    - CommonSecurityLog (Zscaler)   
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  -  PrivilegeEscalation
+relevantTechniques:
+  - T1078
+query: |
+    let Alert_List= dynamic([
+    "Phishing link click observed in Network Traffic", 
+    "Phish delivered due to an IP allow policy",
+    "A potentially malicious URL click was detected",
+    "High Risk Sign-in Observed in Network Traffic",
+    "A user clicked through to a potentially malicious URL",
+    "Suspicious network connection to AitM phishing site", 
+    "Messages containing malicious entity not removed after delivery", 
+    "Email messages containing malicious URL removed after delivery",
+    "Email reported by user as malware or phish",
+    "Phish delivered due to an ETR override",
+    "Phish not zapped because ZAP is disabled"]);
+    SecurityAlert
+    | where AlertName in~ (Alert_List)
+    //Findling Alerts which has the URL
+    | where Entities has "url"
+    //extracting Entities 
+    | extend Entities = parse_json(Entities)
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'account'
+        | extend EntityUPN = iff(isempty(Entity.UserPrincipalName), tostring(strcat(Entity.Name, "@", tostring (Entity.UPNSuffix[0]))), tostring(Entity.UserPrincipalName))
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'ip'
+        | extend EntityIp = tostring(Entity.Address)
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'host'
+        | extend EntityHost = tostring(Entity.HostName)
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'url'
+        | extend EntityUrl = tostring(Entity.Url)
+        )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+        AccountUpn=take_any(EntityUPN),
+        IPAddress=take_any(EntityIp),
+        HostName=take_any(EntityHost),
+        Url=tostring(tolower(take_any(EntityUrl))),
+        AlertTime= min(TimeGenerated),
+        make_set(SystemAlertId)
+        by ProductName, AlertName
+    // matching with 3rd party network logs and 3p Alerts 
+    | join kind= inner (CommonSecurityLog
+        | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
```

Comment:
Consider adding DeviceProduct filter as well

Reply:
Similar to above, filter CommonSecurityLog by time to increase query efficiency.

Reply:
its there 
        | where DeviceProduct startswith "FortiGate" or DeviceProduct startswith  "PAN" or DeviceProduct startswith  "VPN" or DeviceProduct startswith "FireWall" or DeviceProduct startswith  "NSSWeblog" or DeviceProduct startswith "URL"